tgui/packages/tgui/interfaces/common/Objectives.tsx

```diff
@@ -0,0 +1,57 @@
+import { BooleanLike } from 'common/react';
+import { InfernoNode } from 'inferno';
+import { Stack } from '../../components';
+
+export type Objective = {
+  // The title of the objective, not actually displayed so optional
+  name?: string;
+  // What "number" objective this is, IE, its index in the list of objectives
+  count: number;
+  // The text explaining what this objective requires
+  explanation: string;
+  // Whether or not this objective is completed
+  complete: BooleanLike;
+};
+
+export const ObjectivePrintout = (
+  props: {
+    // For passing onto the Stack component
+    fill?: boolean;
+    // Allows additional components to follow the printout in the same stack
+    objectiveFollowup?: InfernoNode;
+    // The prefix to use for each objective, defaults to "#" (#1, #2)
+    objectivePrefix?: string;
+    // The font size to use for each objective
+    objectiveTextSize?: string;
+    // The objectives to print out
+    objectives: Objective[];
+    // The title to use for the printout, defaults to "Your current objectives"
+    titleMessage?: string;
+  },
+  context
+) => {
+  const {
+    fill,
+    objectiveFollowup,
+    objectivePrefix,
+    objectiveTextSize,
+    objectives,
+    titleMessage,
+  } = props;
+
+  return (
+    <Stack fill={fill} vertical>
```

Comment:
Someone let me know if there's a better way of passing props to components like so

Reply:
this is pretty conventional. if you have a lot of fill={fill}'s you can use {...{ fill, something, something }}. i would suggest it here though because of vertical